File: book/스프링부트를 활용한 마이크로서비스 개발/test/src/main/java/com/example/test/repository/MultiplicationResultAttemptRepository.java
package com.example.test.repository;

import com.example.test.domain.MultiplicationResultAttempt;
import org.springframework.data.repository.CrudRepository;

import java.util.List;

public interface MultiplicationResultAttemptRepository
        extends CrudRepository<MultiplicationResultAttempt, Long> {

    List<MultiplicationResultAttempt> findTop5ByUserAliasOrderByIdDesc(String userAlias);
}

File: book/스프링부트를 활용한 마이크로서비스 개발/kotlin/src/main/kotlin/com/example/kotlin/repository/MultiplicationResultAttemptRepository.kt
package com.example.kotlin.repository

import com.example.kotlin.domain.MultiplicationResultAttempt
import org.springframework.data.repository.CrudRepository

interface MultiplicationResultAttemptRepository
    : CrudRepository<MultiplicationResultAttempt, Long> {

        /*
        * @return 닉네임에 해당하는 사용자의 최근 답안 5개
        * */
        fun findTop5ByUserAliasOrderByIdDesc(userAlias: String): List<MultiplicationResultAttempt>
}